### ISSUE TITLE
Update logs for VAE example for overriding train_step

### ISSUE THREAD

**Issue by siavash-khodadadeh** [keras-team/keras-io#48]
I am learning how to override train_step from the guides. It seems to me that instead of returning the values here:
https://github.com/keras-team/keras-io/blob/master/examples/generative/vae.py#L91

We need to update some metrics like this:
```
self.loss_metric.update_state(total_loss)
self.reconstruction_loss_metric.update_state(reconstruction_loss)
self.kl_loss_metric.update_state(kl_loss)

return {
            "loss": total_loss,
            "reconstruction_loss": reconstruction_loss,
            "kl_loss": kl_loss,
        }
```
I might be wrong, but I think if we do not do this, the model.fit with verbose option just prints the loss for the final batch of the epoch instead of the average loss on the whole epoch. I guess it would be great if test_step is also added to and explained in these guides.

**Comment by fchollet** [comment]
You're right, this pattern will only print last-batch metrics. In this case this is not really a problem, though it could deserve a comment.

In the general case, if you want to track the mean of a tensor over time, you could:

- create a `self.mean_loss = metrics.Mean()` object in `__init__`
- in `train_step`, call `self.mean_loss.update_state(loss_value)` then return `self.mean_loss.result()`

**Comment by siavash-khodadadeh** [comment]
I see. We have to return the metric.result(). Thank you for the clarification.
